midend/convertErrors.cpp

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2021 Intel Corporation
```

Comment:
There is an alternative way to do this which you may find simpler: convert the Error to a serializable Enum and the rely on the serializable enum conversion pass. This should be simpler than what you have here.